src/python/pants/engine/internals/graph.py

```diff
@@ -1206,7 +1207,7 @@ def __init__(
 
 @rule
 async def determine_explicitly_provided_dependencies(
-    request: DependenciesRequest,
+    request: DependenciesRequest,  # NB: This rule ignores request.should_traverse_deps_predicate.
```

Comment:
We might consider a different input type that doesn't have the ignored attribute?

Reply:
Addressed in #19272